src/lib/storage.test.ts

```diff
@@ -2,9 +2,91 @@
 // SPDX-FileCopyrightText: 2023-Present The Pepr Authors
 
 import { beforeEach, describe, expect, it, jest } from "@jest/globals";
+import { DataStore, Storage, v2StoreKey, stripV2Prefix } from "./storage";
+import fc from "fast-check";
+
+describe("stripV2Prefix", () => {
+  it("should remove the v2 prefix", () => {
+    const keys = ["v2-key1", "v2-key2", "v2-key3", "v2-key4", "v2-key5"];
+    const results = ["key1", "key2", "key3", "key4", "key5"];
+
+    for (let i = 0; i < keys.length; i++) {
+      const result = stripV2Prefix(keys[i]);
+      expect(result).toEqual(results[i]);
+    }
+  });
+});
+describe("v2StoreKey", () => {
+  it("should prefix the key with v2", () => {
+    const keys = ["key1", "key2", "key3", "key4", "key5"];
+    const results = ["v2-key1", "v2-key2", "v2-key3", "v2-key4", "v2-key5"];
+
+    for (let i = 0; i < keys.length; i++) {
+      const result = v2StoreKey(keys[i]);
+      expect(result).toEqual(results[i]);
+    }
+  });
+});
+describe("Storage with fuzzing and property-based tests", () => {
+  let storage: Storage;
 
-import { DataStore, Storage } from "./storage";
+  beforeEach(() => {
+    storage = new Storage();
+    storage.registerSender(jest.fn());
+  });
+
+  it("should correctly set and retrieve items", () => {
+    fc.assert(
+      fc.property(fc.string(), fc.string(), (key, value) => {
+        storage.setItem(key, value);
+        const mockData: DataStore = { [v2StoreKey(key)]: value };
+        storage.receive(mockData);
+        if (value === "") {
+          expect(storage.getItem(key)).toBeNull();
+        } else {
+          expect(storage.getItem(key)).toEqual(value);
+        }
+      }),
+      { numRuns: 100 },
+    );
+  });
 
+  it("should return null for non-existing items", () => {
+    fc.assert(
+      fc.property(fc.string(), key => {
+        expect(storage.getItem(key)).toBeNull();
+      }),
+      { numRuns: 100 },
+    );
+  });
+
+  it("should correctly remove items", () => {
+    fc.assert(
+      fc.property(fc.string(), fc.string(), (key, value) => {
+        storage.setItem(key, value);
+        storage.removeItem(key);
+        expect(storage.getItem(key)).toBeNull();
+      }),
+      { numRuns: 100 },
+    );
+  });
+
+  it("should ensure all set items are base64 encoded internally", () => {
```

Comment:
```suggestion
  it("should ensure all set items are v2-coded internally", () => {
```

Fix hold-over from previous impl..?